File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/data/network/api/RideApi.java
package com.lattis.ellipse.data.network.api;


import com.lattis.ellipse.data.network.model.body.ride.EndRideBody;
import com.lattis.ellipse.data.network.model.body.ride.RideRatingBody;
import com.lattis.ellipse.data.network.model.body.ride.RideSummaryBody;
import com.lattis.ellipse.data.network.model.body.ride.StartRideBody;
import com.lattis.ellipse.data.network.model.body.ride.UpdateRideBody;
import com.lattis.ellipse.data.network.model.response.history.RideHistoryResponse;
import com.lattis.ellipse.data.network.model.response.ride.RideSummaryResponse;
import com.lattis.ellipse.data.network.model.response.ride.StartRideResponse;
import com.lattis.ellipse.data.network.model.response.ride.UpdateTripResponse;

import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import io.reactivex.Observable;

public interface RideApi {

    @POST("trips/start-trip/")
    Observable<StartRideResponse> startRide(@Body StartRideBody startRideBody);

    @POST("trips/end-trip/")
    Observable<Void> endRide(@Body EndRideBody endRideBody);

    @POST("trips/update-trip/")
    Observable<UpdateTripResponse> updateRide(@Body UpdateRideBody updateRideBody);


    @POST("trips/get-trip-details/")
    Observable<RideSummaryResponse> getRideSummary(@Body RideSummaryBody rideSummaryBody);

    @POST("trips/update-rating/")
    Observable<Void> rateRide(@Body RideRatingBody rideRatingBody);

    @GET("trips/get-trips/")
    Observable<RideHistoryResponse> getRideHistory();

}

File: velolabs_repos/lattis_android_v2/data/src/main/java/com/lattis/data/repository/datasources/api/ride/RideApi.kt
package com.lattis.data.repository.datasources.api.ride

import com.lattis.data.entity.body.ride.*
import com.lattis.data.entity.response.history.RideHistoryResponse
import com.lattis.data.entity.response.ride.*
import com.lattis.domain.models.RideHistory
import io.reactivex.rxjava3.core.Observable
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST

interface RideApi {
    @POST("api/trips/get-trip-details/")
    fun rideSummary(@Body rideSummaryBody: RideSummaryBody): Observable<RideSummaryResponse>

    @POST("api/trips/update-trip/")
    fun updateRide(@Body updateRideBody: UpdateRideBody): Observable<UpdateTripResponse>

    @POST("api/trips/start-trip/")
    fun startRide(@Body startRideBody: StartRideBody): Observable<StartRideResponse>


    @POST("api/trips/end-trip/")
    fun endRide(@Body endRideBody: EndRideBody): Observable<EndRideResponse>

    @POST("api/trips/update-rating/")
    fun rateRide(@Body rideRatingBody: RideRatingBody): Observable<RideRatingResponse>

    @GET("api/trips/get-trips/")
    fun getRideHistory(): Observable<RideHistory>
}